edgeless_node/src/agent/mod.rs

```diff
@@ -65,33 +67,104 @@ impl Agent {
         // value: provider_id
         let mut resource_instances = std::collections::HashMap::<edgeless_api::function_instance::ComponentId, String>::new();
 
+        // After spawning a new function, the function´s class is only used to determine which runner to start it on.
+        // When stopping, only the stop_function_id is provided which does not allow to know which runner it is
+        // currently deployed on. Here, we implement a instance_id -> function_class HashMap
+        let mut component_id_to_class_map = std::collections::HashMap::<edgeless_api::function_instance::ComponentId, String>::new();
```

Comment:
Probably we should not duplicate the string here (use an index/enum).
But this is an optimization that can be added later.

Reply:
Fair aspect, might increase lookup time and complexity of the overall code for a fairly small memory optimization though